avaframe/in4Region/splitInputs.py

```diff
@@ -0,0 +1,296 @@
+
+    # Step 1: Create the central list
+    log.info("Creating folder list...")
+    folderList = createFolderList(inputShp)
+    # Group folders with identical "name" attributes before the first underscore and update the list
+    folderListGrouped = groupFoldersByName(folderList)
+    log.info("Finished creating folder list")
+
+    # Step 2: Set up ava directories
+    log.info("Running folder initialization for each entry...")
+    for entry in folderListGrouped:
+        folderName = entry['folderName']
+        initializeFolderStruct(str(outputDir / folderName), removeExisting=True)
+        log.debug(f"Created folder structure for '{folderName}'.")
+    log.info("Finished folder initialization")
+
+    # Step 3: Split and move release areas to each directory
+    log.info("Splitting and moving release areas...")
+    splitAndMoveReleaseAreas(folderListGrouped, inputShp, outputDir)
+    log.info("Finished splitting and moving release areas")
+
+    # Step 4: Clip and move DEM
+    if cfg['GENERAL'].getboolean('splitDEM'):
+        log.info("Clipping and moving DEM...")
+        clipDEMByCentroidAndMove(folderListGrouped, inputDEM, outputDir, cfg)
+        log.info("Finished clipping and moving of DEM")
+
+    # (Step 5: Clip and move ENT and RES... other input too?)
+
+    # Step 6: Divide release areas into scenarios based on "scenario" attribute, if it exists
+    log.info("Separating release area by scenario attribute...")
+    splitByScenarios(folderListGrouped, outputDir)
+    log.info("Finished separating by scenarios")
+
+def readShapefile(src):
+    """Read the fields, properties, and geometries of an input shapefile.
+    To be used in combination with shapefile.Reader. Could be expanded upon to get e.g.
+    the srs, shapeTypes, bounds, numFeatures and metadata if needed
+
+    Parameters
+    ----------
+    src: shapefile.Reader
+        the input shapefile
+
+    Returns
+    -------
+    fields: list
+        the fields of the input shapefile
+    fieldNames: list
+        the names of the fields
+    properties: list
+        a list of dictionaries containing the properties of each feature
+    geometries: list
+        a list of geometry objects
+    """
+
+    fields = src.fields[1:]  # Skip deletion flag
+    fieldNames = [field[0] for field in fields]
+    properties = []
+    geometries = []
+    for shapeRecord in src.iterShapeRecords():
+        properties.append(dict(zip(fieldNames, shapeRecord.record)))
+        geometries.append(shape(shapeRecord.shape.__geo_interface__))
+    return fields, fieldNames, properties, geometries
+
+def createFolderList(inputShp):
+    """Create a list of entries of each feature in the input shapefile."""
+    folderList = []
+    unnamedCount = 1
+    with shapefile.Reader(inputShp) as src:
+        fields, fieldNames, properties, geometries = readShapefile(src)
+
+        for i, (properties, geometry) in enumerate(zip(properties, geometries)):
```

Comment:
This for loop does not need to live within the with, right? So it could be moved forward